System/ContentFile.h

```diff
@@ -32,483 +19,253 @@ struct ga_Handle;
+		/// </summary>
+		/// <returns>Whether the data of this has been modified in memory since load.</returns>
+		//bool IsDataModified() const { return m_DataModified; };
+
+		/// <summary>
+		/// Tells this whether the data loaded by this into memory has been modified and should be saved out to this' path upon writing.
+		/// </summary>
+		/// <param name="modified">Whether the loaded data of this has been modified or not.</param>
+		//virtual void SetDataModified(bool modified = true) { m_DataModified = modified; };
+
+		/// <summary>
+		/// Creates a hash value out of a path to a ContentFile.
+		/// </summary>
+		/// <returns>Hash value of a path to a ContentFile.</returns>
+		virtual size_t GetHash() const { return std::hash<std::string>()(m_DataPath); }
+
+		/// <summary>
+		/// Converts hash values into file paths to ContentFiles.
+		/// </summary>
+		/// <param name="hash">Hash value to get file path from.</param>
+		/// <returns>Path to ContentFile.</returns>
+		static std::string GetPathFromHash(size_t hash);
+#pragma endregion
+
+#pragma region Data Handling
+		/// <summary>
+		/// Loads and gets the data represented by this ConentFile object as an Allegro BITMAP. Note that ownership of the BITMAP IS NOT TRANSFERRED!
+		/// Also, this should only be done once upon startup, since getting the BITMAP again is slow.
+		/// </summary>
+		/// <param name="conversionMode">
+		/// The Allegro color conversion mode to use when loading this bitmap.
+		/// Note it will only apply the first time you get a bitmap since it is only loaded from disk the first time. See allegro docs for the modes.
+		/// </param>
+		/// <returns>The pointer to the beginning of the data object loaded from the allegro .dat datafile. file. Ownership is NOT transferred! If 0, the file could not be found/loaded.</returns>
+		virtual BITMAP * GetAsBitmap(int conversionMode = 0);
+
+		/// <summary>
+		/// Loads and gets the data represented by this ConentFile object as an of array Allegro BITMAPs, each representing a frame in an animation.
+		/// Note that ownership of the BITMAPS ARE NOT TRANSFERRED, BUT THE ARRAY ITSELF, IS!
+		/// Also, this should only be done once upon startup, since getting the BITMAPs again is slow.
+		/// </summary>
+		/// <param name="frameCount">
+		/// The number of frames to attempt to load.
+		/// The name of the file/object specified by the data path will be appended with 000, 001. 002 etc up to the frameCount number - 1.
+		/// </param>
+		/// <param name="conversionMode">
+		/// The Allegro color conversion mode to use when loading this bitmap.
+		/// Note it will only apply the first time you get a bitmap since it is only loaded from disk the first time. See allegro docs for the modes.
+		/// </param>
+		/// <returns>
+		/// The pointer to the beginning of the array of BITMAP pointers loaded from the allegro .dat datafile. file, the length of which is specified with the frameCount argument.
```

Comment:
Typo - ".dat datafile. file, the length" should be ".dat datafile, the length"
Also, I would move the last bit about If 0, etc. to a new line since it's a new concept.